core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -80,8 +86,59 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
   //periodically emit metrics (don't need to do this for each message!)
   context.system.scheduler.schedule(30.seconds, 10.seconds, self, EmitMetrics)
 
+  // Key is `kind,memory,time`, value is the number of cold Start in minute
+  var coldStartCount = immutable.Map.empty[String, Int]
+
   backfillPrewarms(true)
 
+  // check periodically every 1 minute, delete prewarmed container if unused for sometime
+  context.system.scheduler.schedule(10.seconds, 1.minute, self, DeleteUnusedPrewarmedContainer)
```

Comment:
So the main idea is removing unused prewarm containers every 1 minute and create more prewarm containers based on the usage during the last 1 minute.

I think it is a good starting point.
This is a kind of heuristic approach expecting the future based on history

One thing to consider is, we can't exactly expect the required prewarm types.
Any kind of activation can come at any time.
If an action is sparsely invoked or invoked with a bigger interval than prewarm TTL, it will always trigger a cold-start.
So how about introducing a minimum limit as well?
We can configure the minimum number of prewarm containers while controlling the ratio of them based on the usage.

Reply:
Regarding `So how about introducing a minimum limit as well?`
You mean add `minCount` in runtime.json

Regarding `We can configure the minimum number of prewarm containers while controlling the ratio of them based on the usage.`
What's mean? the minimum numbe can be changed on some cases?

Reply:
It is true that without a minimum, the prewarms may all go to zero, which is what you need if you have blackbox dedicated invokers if you don't want any prewarms running there.
So the stemcell config could have:
initialCount : this should probably be whatever the value is used today for number of prewarms of that config.
minCount : 0 is required, if you want to prevent case where unused prewarms are wasting resources, but it implies that you will get a cold start for that config on that invoker.
maxCount : this is an upper limit on max number of prewarms for that config.

Reply:
For future work, I would also consider:
- Does this invoker support only blackbox? If so, it can safely scale down to 0 prewarms, since blackbox are never prewarmed.
- Can new prewarm configs be completely inferred? So that if lots of actions of a particular config arrive, they eventually get benefits of having prewarms.
- Can blackbox then be allowed to have prewarms, if they receive enough traffic at this invoker?

Reply:
Already added `minCount` in runtimes.json